tests/leds/Makefile

```diff
@@ -1,3 +1,6 @@
 include ../Makefile.tests_common
 
+# Some boards do not initialize LED0 by default
+CFLAGS=-DAUTO_INIT_LED0
```

Comment:
I'm not sure about this PR, or better about the usage (and usefulness) of the `AUTO_INIT_LED0` macro in general.
Because it is not consistently used for all boards, a close look reveals it is mostly used by the nucleos but most other boards don't have it

```
git grep AUTO_INIT_LED0 
boards/b-l072z-lrwan1/board.c:#ifdef AUTO_INIT_LED0
boards/b-l072z-lrwan1/include/gpio_params.h:#ifdef AUTO_INIT_LED0
boards/b-l475e-iot01a/board.c:#ifdef AUTO_INIT_LED0
boards/b-l475e-iot01a/include/gpio_params.h:#ifdef AUTO_INIT_LED0
boards/common/nucleo/board.c:#ifdef AUTO_INIT_LED0
boards/common/nucleo144/include/board.h:#define AUTO_INIT_LED0
boards/common/nucleo32/include/gpio_params.h:#ifndef AUTO_INIT_LED0
boards/common/nucleo32/include/gpio_params.h:#endif /* AUTO_INIT_LED0 */
boards/common/nucleo64/include/gpio_params.h:#ifdef AUTO_INIT_LED0
boards/lobaro-lorabox/include/board.h:#define AUTO_INIT_LED0
boards/lobaro-lorabox/include/gpio_params.h:#ifdef AUTO_INIT_LED0
boards/sodaq-sara-aff/board.c:    #ifdef AUTO_INIT_LED0
```

To me it looks like we should either remove its usage completely or specify (and implements) its usage properly and for all boards consistently.

Reply:
`AUTO_INIT_LED0` is a fix for nucleo based boards, because there, LED and SPI pins conflict. With the flag set, we explicitly enable LED over SPI. Maybe we should document it better. However, IMO enabling the flag for this test is the right thing to do. Follow up PRs should correct behavior on non-nucleo boards (lobaro already fixed in  #11060).

Reply:
yeah I get it now, sorry about the noise - still in the long run we should come up with a better solution than this inconsistent macro usage all over the place ...